imagetest/test_suites/metadata/setup.go

```diff
@@ -12,40 +12,189 @@ var Name = "metadata"
 
 const (
 	daemonScriptTemplate = `#!/bin/bash
-
-nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
-echo $! > %s
-`
+	nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
+	echo $! > %s
+	`
 	daemonOutputPath = "/daemon_out.txt"
 
-	startupScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	startupOutputPath  = "/startup_out.txt"
-	startupContent     = "The startup script worked."
-	shutdownScriptTime = `#!/bin/bash
-
-while [[ 1 ]]; do
-  date +%s >> /shutdown.txt
-  sync
-  sleep 1
-done`
-	shutdownScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	shutdownOutputPath = "/shutdown_out.txt"
-	shutdownContent    = "The shutdown script worked."
+	bashScriptTemplate = `#!/bin/bash
+	echo "%s" > %s`
+	windowsScriptTemplate = `echo "%s" > %s
+	ping -n 2 127.0.0.1 >NUL`
+	powershellScriptTemplate = `Set-Content -Value "%s" -Path %s
+	Start-Sleep 1`
+	startupContent  = "The startup script worked."
+	shutdownContent = "The shutdown script worked."
+
+	timeOutputPath        = "/shutdown.txt"
+	windowsTimeOutputPath = "C:\\shutdown.txt"
+
+	bashTimeScriptTemplate = `#!/bin/bash
+	while [[ 1 ]]; do
+	date +%%s >> %s
+	  sync
+	  sleep 1
+	  done`
+
+	windowsTimeScriptTemplate = `while ($true) {
+		$time = (Get-Date).ToString("yyyyMMdd-HHMMss")
+		Add-Content -Path %s -Value $time
```

Comment:
this is only used in one place, why make this a pass-in value? the naming for this script should also indicate it is specific to shutdown scripts

Reply:
Like the previous one, while the script itself is only used once, the path value is used more than once.